platforms/cuda/src/CudaKernels.cpp

```diff
@@ -1650,14 +1668,29 @@ void CudaCalcNonbondedForceKernel::initialize(const System& system, const Nonbon
     // Initialize nonbonded interactions.
 
     int numParticles = force.getNumParticles();
+    // Pack the C6 coeffiecient in with sigma and epsilon, in case LJPME is being used. The C6
+    // coefficients could live in a separate array, but that would hurt cache efficiency for LJPME.
+    doLJPME = nonbondedMethod == LJPME;
     sigmaEpsilon = CudaArray::create<float2>(cu, cu.getPaddedNumAtoms(), "sigmaEpsilon");
+    if(doLJPME){
+        if (cu.getUseDoublePrecision())
+            C6s = CudaArray::create<double>(cu, cu.getPaddedNumAtoms(), "C6s");
+        else
+            C6s = CudaArray::create<float>(cu, cu.getPaddedNumAtoms(), "C6s");
+    }
     CudaArray& posq = cu.getPosq();
     vector<double4> temp(posq.getSize());
     float4* posqf = (float4*) &temp[0];
     double4* posqd = (double4*) &temp[0];
+    // The C6 coefficients for LJPME could be computed from sigma and epsilon, but it
+    // seems like a good idea to cache them and avoid many recomputations.
```

Comment:
On a GPU, memory access is much more expensive than computation.  If it takes less than a few hundred flops to compute something, that's probably faster than loading it from memory.